src/entities/addLiquidityBoosted/index.ts

```diff
@@ -0,0 +1,187 @@
+// A user can add liquidity to a boosted pool in various forms. The following ways are
+// available:
+// 1. Unbalanced - addLiquidityUnbalancedToERC4626Pool
+// 2. Proportional - addLiquidityProportionalToERC4626Pool
+import { TokenAmount } from '@/entities/tokenAmount';
+
+import { Permit2 } from '@/entities/permit2Helper';
+
+import { getAmountsCall } from '../addLiquidity/helpers';
+
+import { PoolState, PoolStateWithUnderlyings } from '@/entities/types';
+
+import {
+    AddLiquidityBaseBuildCallInput,
+    AddLiquidityBaseQueryOutput,
+    AddLiquidityBuildCallOutput,
+    AddLiquidityInput,
+    AddLiquidityKind,
+} from '../addLiquidity/types';
+
+import { doAddLiquidityUnbalancedQuery } from './doAddLiquidityUnbalancedQuery';
+import { doAddLiquidityProportionalQuery } from './doAddLiquidityPropotionalQuery';
+import { Token } from '../token';
+import { BALANCER_COMPOSITE_LIQUIDITY_ROUTER } from '@/utils';
+
+import { getValue } from '@/entities/utils/getValue';
+import { encodeFunctionData } from 'viem';
+import { balancerCompositeLiquidityRouterAbi, balancerRouterAbi } from '@/abi';
+
+import { Hex } from '@/types';
+
+export class AddLiquidityBoostedV3 {
+    async query(
+        input: AddLiquidityInput,
+        poolState: PoolState | PoolStateWithUnderlyings,
```

Comment:
Seems like there is something incorrect here? I don't think we can use `PoolState`?

Reply:
The required state was actually dependend on the type. I now require to simply always pass a `PoolStateWithUnderlyings`. Only in the proportional adds was the type with underlying requires as the returned TokenAmounts need to have the pool's token underlyings but in the case of an unbalanced the returned TokenAmount is the BPT and all information was already shared in the input/poolstate.